src/main/src/test/java/org/geoserver/catalog/impl/CatalogImplTest.java

```diff
@@ -2149,6 +2184,31 @@ public void testGetLayerGroupByName() {
         assertNull(catalog.getLayerGroupByName("cite", "layerGroup2"));
     }
 
+    @Test
+    public void testRemoveLayerGroupAndAssociatedDataRules() throws IOException {
+        DataAccessRuleDAO dao = DataAccessRuleDAO.get();
+        CatalogListener listener = new SecuredResourceNameChangeListener(catalog, dao);
+        // catalog.getListeners().add()
```

Comment:
?

Reply:
do you mean that is not necessary to test layer group also?